Closing the Loop - VS

KrisNotChris1 opened the pull request:
Added rounds to the matches.
Added a win condition for the matches.
Added animations from Ryland for Round 2, Final Round, Player 1 Wins, and Player 2 Wins.

KrisNotChris1 committed:
Closing the Loop - VS

Added rounds to the matches.
Added a win condition for the matches.
Added animations from Ryland for Round 2, Final Round, Player 1 Wins, and Player 2 Wins.

johnkiddjr commented on scripts/setupgame/setupgame.gml:
```diff
@@ -110,12 +110,31 @@ function SetupGame()
 				{
-					sprite_index = sRound1Start;
-					lifetime = 110;
+					if(global.currentRound = 1)
```
Use ```==``` for equivalence, instead of ```=```

johnkiddjr commented on scripts/setupgame/setupgame.gml:
```diff
@@ -110,12 +110,31 @@ function SetupGame()
+						sprite_index = sRound1Start;
+						lifetime = 110;
+					}
+					if(global.currentRound = 2)
```
Use ```==``` for equivalence, instead of ```=```

This should also be ```else if``` since the current round cannot be 1, 2, and 3 simultaneously.

johnkiddjr commented on scripts/setupgame/setupgame.gml:
```diff
@@ -110,12 +110,31 @@ function SetupGame()
+						sprite_index = sRound2Start;
+						lifetime = 110;
+					}
+					if(global.currentRound = 3)
```
Use ```==``` for equivalence, instead of ```=```

This should also be ```else if``` since the current round cannot be 1, 2, and 3 simultaneously.

johnkiddjr commented on objects/ogamemanager/Step_0.gml:
```diff
@@ -76,6 +76,42 @@ switch (global.gameMode)
 		}
+		
+		// When a player meets the win requirement for the match return players to the character selection screen
+		if (global.p1Rounds == 2 || global.p2Rounds == 2)
```
This needs to be rewritten so the same condition isn't checked twice. You'll have to repeat 2 more lines, but it'll be more efficient with cpu resources.

```
if (global.p1Rounds == 2)
{
   global.gameHalt = true;
   gameHaltTimer++;
   var particle = instance_create_layer(global.camObj.x - 80, 0, "KO_Text", oParticles);
   //... etc
}
else if (global.p2Rounds == 2)
{
   global.gameHalt = true;
   gameHaltTimer++;
   var particle = instance_create_layer(global.camObj.x - 80, 0, "KO_Text", oParticles);
    // etc...
}
```

Alternatively, you could pull everything out into a function and just have the if call the function to reduce code repeat.

Something like this:

```
if (global.p1Rounds == 2)
{
    gameHaltTimer++;
    PlayerWins(1);
}
else if (global.p2Rounds == 2)
{
    gameHaltTimer++;
    PlayerWins(2);
}

// function
function PlayerWins(playerNumber)
{
    global.gameHalt = true;
    ...
    sprite_index = asset_get_index("sPlayer" + playerNumber + "Wins");
}
```

KrisNotChris1 committed:
Close the Loop-VS- 2

Fixed style issues
Figured out how to stop the game from pausing at the beginning of new rounds (Round 2 and beyond)
